marklogic-client-api/src/test/java/com/marklogic/client/test/ZipSplitterTest.java

```diff
@@ -0,0 +1,72 @@
+package com.marklogic.client.test;
+
+import com.marklogic.client.datamovement.ZipSplitter;
+import com.marklogic.client.document.DocumentWriteOperation;
+import com.marklogic.client.io.BytesHandle;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Iterator;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static org.junit.Assert.*;
+
+public class ZipSplitterTest {
+    static final private String zipFile = "src/test/resources/data" + File.separator + "files.zip";
+
+    @Test
+    public void testSplitter() throws Exception {
+
+        ZipSplitter splitter = new ZipSplitter();
+        Stream<BytesHandle> contentStream = splitter.split(new ZipInputStream(new FileInputStream(zipFile)));
+        assertNotNull(contentStream);
+
+        BytesHandle[] bytesResult = contentStream.toArray(size -> new BytesHandle[size]);
+        assertNotNull(bytesResult);
+
+        ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(zipFile));
+        ZipEntry zipEntry = null;
+
+        for (int i = 0; (zipEntry = zipInputStream.getNextEntry()) != null && i < bytesResult.length; i++) {
+            assertNotNull(bytesResult[i].get());
+            checkContent(zipInputStream, zipEntry, new String(bytesResult[i].get()));
+        }
+    }
+
+    @Test
+    public void testSplitterWrite() throws Exception {
+
+        ZipSplitter splitter = new ZipSplitter();
+        Stream<DocumentWriteOperation> contentStream =
+                splitter.splitWriteOperations(new ZipInputStream(new FileInputStream(zipFile)));
+        assertNotNull(contentStream);
+
+        ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(zipFile));
+        ZipEntry zipEntry = null;
+
+        Iterator<DocumentWriteOperation> itr = contentStream.iterator();
+        while (itr.hasNext() && ((zipEntry = zipInputStream.getNextEntry()) != null)) {
+            DocumentWriteOperation docOp = itr.next();
+            assertNotNull(docOp.getUri());
+            assertEquals(docOp.getUri(), zipEntry.getName());
+
+            assertNotNull(docOp.getContent());
+            String docOpContent = docOp.getContent().toString();
+            checkContent(zipInputStream, zipEntry, docOpContent);
+        }
+    }
+
```

Comment:
Should we add a filter lambda test on split() and a uriTransformer test on splitWriteOperations()?

Just to make sure that the lambdas are wired in correctly?

Reply:
Sure. Added an entryFilter to test the split method and a uriTransformer to test splitWriteOperations method.

splitter.setEntryFilter(x -> x.getSize() > 50 ? true : false );
splitter.setUriTransformer(name -> name.toUpperCase());